Cleanup Transport*VotingConfigExclusionsActionTests

alyokaz opened the pull request:
This pull request fixes #98057 

The assertions have been moved into the handlers where they can be, and the calls to `countDownLatch.await()` as been replaced by `safeAwait.`

Attempting to replace `TransportResponseHandler` with `ActionListenerResponseHandler` seems to be less concise due to the need to implement an `ActionListener` for its constructor, which then requires the introduction of a new method to keep things DRY.

DaveCTurner commented on server/src/test/java/org/elasticsearch/action/admin/cluster/configuration/TransportAddVotingConfigExclusionsActionTests.java:
```diff
@@ -170,11 +168,11 @@ public void testWithdrawsVoteFromANode() throws InterruptedException {
 
-        assertTrue(countDownLatch.await(30, TimeUnit.SECONDS));
+        safeAwait(countDownLatch);
         assertThat(clusterService.getClusterApplierService().state().getVotingConfigExclusions(), contains(otherNode1Exclusion));
```
You've moved some of the assertions about the current state into the success handler - I would slightly prefer them all to move, but whichever way we go I'd rather we were consistent.

DaveCTurner replied:
This comment is still outstanding.

alyokaz replied:
Apologies, I had just taken a quick look at the merge conflicts, I hadn't meant to give impression I had addressed the issues you raised. I should have made my intentions more explicit.

I was keeping some of the assertions outside the handler is because I'd, superficially, made the assumption the handler must be being called twice in order to decrement the `CountDownLatch` twice, so the assertion would need to be outside the handler to prevent it being called twice also. I can see this isn't the case now, so have moved the remaining assertions as well.

DaveCTurner reviewed (changes requested):
Thanks for the contribution @alyokaz; I left one comment inline. Also to close #98057 you'd have to do the same thing to `TransportClearVotingConfigExclusionsActionTests`. Finally, would you merge the latest `main` and fix up the compile errors?

DaveCTurner reviewed (dismissed):
Thanks for the updates @alyokaz, however this now doesn't even compile (or pass `./gradlew precommit`).

alyokaz committed:
Refactor Transport(Add/Clear)VotingConfigExclusionsActionTests

Move assertions into success handlers.
Replace call to await on CoundDownLatch with safeAwait.

alyokaz commented:
I've ran `./gradlew precommit`, fixed the violations and rebased the branch on to main.

DaveCTurner commented:
@elasticmachine test this please

DaveCTurner reviewed (approved):
LGTM, thanks for the extra iterations @alyokaz

alyokaz commented:
Happy to contribute @DaveCTurner, thanks for the review
